addon/bootstrap.js

```diff
@@ -59,54 +56,66 @@ class Jsm {
 }
 
 async function startup(addonData, reason) {
-  // addonData: Array [ "id", "version", "installPath", "resourceURI", "instanceID", "webExtension" ]  bootstrap.js:48
   Jsm.import(config.modules);
 
   studyUtils.setup({
     studyName: studyConfig.studyName,
     endings: studyConfig.endings,
-    addon: { id: addonData.id, version: addonData.version },
+    addon: {
+      id: addonData.id,
+      version: addonData.version
+    },
     telemetry: studyConfig.telemetry,
   });
   studyUtils.setLoggingLevel(config.log.studyUtils.level);
-  const variation = await chooseVariation();
-  studyUtils.setVariation(variation);
+  studyUtils.setVariation(studyConfig.variation);
 
   if ((REASONS[reason]) === "ADDON_INSTALL") {
-    studyUtils.firstSeen();  // sends telemetry "enter"
-    const eligible = await config.isEligible(); // addon-specific
+    // Sends telemetry "enter".
+    studyUtils.firstSeen();
+    const eligible = await config.isEligible();
     if (!eligible) {
-      // uses config.endings.ineligible.url if any,
-      // sends UT for "ineligible"
-      // then uninstalls addon
+      // Uses config.endings.ineligible.url if any, sends UT for "ineligible", then uninstalls addon.
       await studyUtils.endStudy({reason: "ineligible"});
       return;
     }
   }
   await studyUtils.startup({reason});
 
   console.log(`info ${JSON.stringify(studyUtils.info())}`);
-  // studyUtils.endStudy("user-disable");
-  let eLTDHomepages = getHomepage();
-  let report = TelemetryRappor.createReport("name", eLTDHomepages);
-  console.log(report);
+
+  let value = HomepageStudy.reportValue(studyUtils.studyName);
+  console.log(value);
+  if (value == null) {
+    studyUtils.endStudy({reason: "incorrect homepage"});
```

Comment:
Done.
`endStudy` ends the study, sends a ping, and uninstall the addon. It doesn't return any value.